server/plugins/irc-events/message.ts

```diff
@@ -174,10 +187,19 @@ export default <IrcEventHandler>function (irc, network) {
 			LinkPrefetch(client, chan, msg, cleanMessage);
 		}
 
+		if (!data.message) {
+			return;
+		}
+
 		chan.pushMessage(client, msg, !msg.self);
 
 		// Do not send notifications if the channel is muted or for messages older than 15 minutes (znc buffer for example)
-		if (!chan.muted && msg.highlight && (!data.time || data.time > Date.now() - 900000)) {
+		if (
+			!chan.muted &&
+			msg.highlight &&
+			(!data.time || data.time > Date.now() - 900000) &&
+			msg.type !== MessageType.TAGMSG
```

Comment:
Dead code, you return early already no?

Plus, by now that conditional is a mess and needs to be done differently I think...

Reply:
Fixed.